File: app/src/main/java/com/joeshuff/dddungeongenerator/screens/HomeListAdapter.java
package com.joeshuff.dddungeongenerator.screens;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.view.ContextThemeWrapper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.recyclerview.widget.RecyclerView;
import com.google.gson.Gson;
import com.joeshuff.dddungeongenerator.util.FirebaseTracker;
import com.joeshuff.dddungeongenerator.GeneratingActivity;
import com.joeshuff.dddungeongenerator.R;
import com.joeshuff.dddungeongenerator.memory.MemoryController;
import com.joeshuff.dddungeongenerator.memory.MemoryGeneration;
import org.json.JSONException;

import java.util.List;

public class HomeListAdapter extends RecyclerView.Adapter<HomeListAdapter.PreviousSearchViewHolder> {

    List<MemoryGeneration> memories;

    public HomeListAdapter(List<MemoryGeneration> memoryGenerationList) {
        memories = memoryGenerationList;
    }

    @NonNull
    @Override
    public PreviousSearchViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new PreviousSearchViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.history_item, viewGroup, false));
    }

    @Override
    public void onBindViewHolder(@NonNull PreviousSearchViewHolder previousSearchViewHolder, int i) {
        previousSearchViewHolder.dungeonSeed.setText(memories.get(i).getSeed());
        previousSearchViewHolder.dungeonGenTime.setText(memories.get(i).getGeneratedAt());
        previousSearchViewHolder.dungeonName.setText(memories.get(i).workOutDungeonName());

        previousSearchViewHolder.deleteButton.setOnClickListener(e -> {
            deletePressed(previousSearchViewHolder.itemView.getContext(), memories.get(i));
        });

        previousSearchViewHolder.itemView.setOnClickListener(e -> {
            clicked(previousSearchViewHolder.itemView.getContext(), memories.get(i));
        });
    }

    public void deletePressed(Context c, MemoryGeneration memoryGeneration) {
        DialogInterface.OnClickListener dialogClickListener = (dialog, which) -> {
            switch (which){
                case DialogInterface.BUTTON_POSITIVE:
                    FirebaseTracker.EVENT(c, "MemoryDelete", "Deleted:" + memoryGeneration.getGeneratedAt());

                    try {
                        MemoryController.removeFromMemory(c, memoryGeneration);
                        memories = MemoryController.getMemory(c);
                        notifyDataSetChanged();
                    } catch (JSONException ec) {

                    }
                    break;

                case DialogInterface.BUTTON_NEGATIVE:
                    //No button clicked
                    break;
            }
        };

        AlertDialog.Builder builder = new AlertDialog.Builder(c);
        builder.setMessage("Are you sure you want to delete this dungeon?")
                .setPositiveButton("Yes", dialogClickListener)
                .setNegativeButton("No", dialogClickListener)
                .show();
    }

    public void clicked(Context context, MemoryGeneration memoryGeneration) {
        Intent generator = new Intent(context, GeneratingActivity.class);

        generator.putExtra("instructions", new Gson().toJson(memoryGeneration));

        context.startActivity(generator);
    }

    @Override
    public int getItemCount() {
        return memories.size();
    }

    public static class PreviousSearchViewHolder extends RecyclerView.ViewHolder {

        TextView dungeonName;
        TextView dungeonGenTime;
        TextView dungeonSeed;

        ImageButton deleteButton;

        public PreviousSearchViewHolder(@NonNull View itemView) {
            super(itemView);

            deleteButton = itemView.findViewById(R.id.deleteButton);
            dungeonName = itemView.findViewById(R.id.dungeonNameField);
            dungeonGenTime = itemView.findViewById(R.id.dungeonTimeGennedField);
            dungeonSeed = itemView.findViewById(R.id.dungeonSeedField);
        }
    }

}

File: app/src/main/java/com/joeshuff/dddungeongenerator/screens/home/HomeListAdapter.kt
package com.joeshuff.dddungeongenerator.screens.home

import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.RecyclerView
import com.google.gson.Gson
import com.joeshuff.dddungeongenerator.R
import com.joeshuff.dddungeongenerator.memory.MemoryController
import com.joeshuff.dddungeongenerator.memory.MemoryGeneration
import com.joeshuff.dddungeongenerator.screens.create.GeneratingActivity
import com.joeshuff.dddungeongenerator.util.FirebaseTracker
import org.json.JSONException

class HomeListAdapter(var memories: List<MemoryGeneration>) : RecyclerView.Adapter<HomeListAdapter.PreviousSearchViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): PreviousSearchViewHolder {
        return PreviousSearchViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.history_item, viewGroup, false))
    }

    override fun onBindViewHolder(previousSearchViewHolder: PreviousSearchViewHolder, i: Int) {
        previousSearchViewHolder.dungeonSeed.text = memories[i].seed
        previousSearchViewHolder.dungeonGenTime.text = memories[i].generatedAt
        previousSearchViewHolder.dungeonName.text = memories[i].workOutDungeonName()
        previousSearchViewHolder.deleteButton.setOnClickListener { deletePressed(previousSearchViewHolder.itemView.context, memories[i]) }
        previousSearchViewHolder.itemView.setOnClickListener { clicked(previousSearchViewHolder.itemView.context, memories[i]) }
    }

    fun deletePressed(c: Context, memoryGeneration: MemoryGeneration) {
        val dialogClickListener = DialogInterface.OnClickListener { dialog: DialogInterface?, which: Int ->
            when (which) {
                DialogInterface.BUTTON_POSITIVE -> {
                    FirebaseTracker.EVENT(c, "MemoryDelete", "Deleted:" + memoryGeneration.generatedAt)
                    try {
                        MemoryController.removeFromMemory(c, memoryGeneration)
                        memories = MemoryController.getMemory(c)
                        notifyDataSetChanged()
                    } catch (ec: JSONException) {
                    }
                }
                DialogInterface.BUTTON_NEGATIVE -> {
                }
            }
        }

        val builder = AlertDialog.Builder(c)
        builder.setMessage("Are you sure you want to delete this dungeon?")
                .setPositiveButton("Yes", dialogClickListener)
                .setNegativeButton("No", dialogClickListener)
                .show()
    }

    fun clicked(context: Context, memoryGeneration: MemoryGeneration?) {
        val generator = Intent(context, GeneratingActivity::class.java)
        generator.putExtra("instructions", Gson().toJson(memoryGeneration))
        context.startActivity(generator)
    }

    override fun getItemCount(): Int {
        return memories.size
    }

    class PreviousSearchViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var dungeonName: TextView
        var dungeonGenTime: TextView
        var dungeonSeed: TextView
        var deleteButton: ImageButton

        init {
            deleteButton = itemView.findViewById(R.id.deleteButton)
            dungeonName = itemView.findViewById(R.id.dungeonNameField)
            dungeonGenTime = itemView.findViewById(R.id.dungeonTimeGennedField)
            dungeonSeed = itemView.findViewById(R.id.dungeonSeedField)
        }
    }

}